controllers/managedocs_controller.go

```diff
@@ -183,6 +193,14 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			},
 		),
 	)
+	servicePredicate := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(client client.Object) bool {
+				name := client.GetName()
+				return name == prometheusServiceName
+			},
+		),
+	)
```

Comment:
You don't need this, the `Owns` below will listen to any service which has a controller reference (for this controller) on it. Manual watches (with selectors/predicates) are just for non-owned resources.

Reply:
I changed it as said, but one question. In this way it will reconcile for any service being created (doesn't matter what name the service has) in the cluster, right?. This goes same with secrets as well (as we own secrets also).
Don't we have to say that reconcile for Service with this particular name (as we know that is Secret we created) and we don't have to reconcile for a user created secret (as we don't take any action with XYZ secret)

Reply:
> I changed it as said, but one question. In this way it will reconcile for any service being created (doesn't matter what name the service has) in the cluster, right?.

No that is not correct. It will only reconcile for services that have an owner reference for the CR managed by this controller, and the only one that matches this condition is the service CR you have added.